core/src/main/java/org/apache/calcite/adapter/enumerable/EnumerableFilter.java

```diff
@@ -68,4 +75,29 @@ public Result implement(EnumerableRelImplementor implementor, Prefer pref) {
     // EnumerableCalc is always better
     throw new UnsupportedOperationException();
   }
+
+  @Override public Pair<RelTraitSet, List<RelTraitSet>> passThroughTraits(
+      RelTraitSet required) {
+    RelCollation collation = required.getCollation();
+    if (collation == null || collation.getFieldCollations().size() == 0) {
+      return null;
+    }
+
+    // Filter, not like Project, does not change on which column that collation defines.
+    return Pair.of(required, ImmutableList.of(required));
+  }
+
+  @Override public Pair<RelTraitSet, List<RelTraitSet>> deriveTraits(
+      final RelTraitSet childTraits, final int childId) {
+    RelCollation collation = childTraits.getCollation();
+    if (collation == null || collation.getFieldCollations().size() == 0) {
```

Comment:
I think the` size == 0` check is not necessary. Did you try remove it?

Reply:
There were empty collation cases (surprisingly but there were some).

That's why I have to check this(otherwise later it will fail)

Reply:
Ok I tried to remove all `size()==0` and it works. 

So i will remove such checks. (It's still wired to me but will let it go).

Reply:
In OptimizeTask, it will call `equalsSansConvention` before derive traits, so I guess it won't come here. Unless some other tests use Enumerable to test distribution trait, in which case, collation might be empty.